cmd/statusd/main.go

```diff
@@ -168,6 +169,22 @@ func main() {
 		return
 	}
 
+	if signalsServerAddress != nil && *signalsServerAddress != "" {
+		signalsServer := NewSignalsServer()
+		signalsServer.Setup()
+		err = signalsServer.Listen(*signalsServerAddress)
+		if err != nil {
+			logger.Error("failed to start signals server", "error", err)
+			return
+		}
+		log.Info("Signals server started", "address", signalsServer.Address())
+		defer func() {
+			ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+			defer cancel()
+			signalsServer.Stop(ctx)
```

Comment:
Hmm, actually nothing like that 😄 
It will not be garbage-collected, because there's a HTTP server running here:
https://github.com/status-im/status-go/blob/a830f8b85629c6b79266762f9431a1191f3ef6fa/cmd/statusd/signals_server.go#L70-L75

And the way to stop it is to call `Shutdown`, which I'm doing here inside `defer`

Reply:
You are right, I missed this yet another block with reference to the server :+1: